shared/src/commonTest/kotlin/co/touchlab/kampstarter/BreedModelTest.kt

```diff
@@ -13,48 +15,71 @@ abstract class BreedModelTest {
     private val settings = MockSettings()
     private val ktorApi = KtorApiMock()
 
+    private var itemDataSummary = CompletableDeferred<ItemDataSummary>()
+    private var errorString = CompletableDeferred<String>()
+
     @BeforeTest
-    fun setup() {
-        TestingServiceRegistry.appStart(dbHelper, settings, ktorApi)
+    fun setup() = runTest {
+        appStart(dbHelper, settings, ktorApi)
+        dbHelper.deleteAll()
 
-        model = BreedModel(viewUpdate = { _ ->
-            // TODO: Test callback invocation
-        }, errorUpdate = { _ ->
-            // TODO: Test callback invocation
+        model = BreedModel(viewUpdate = { summary ->
+            if (summary.allItems.isNotEmpty())
+                itemDataSummary.complete(summary)
+        }, errorUpdate = { s ->
+            errorString.complete(s)
         })
     }
 
     @Test
-    fun `Get Breed List Failure`() = runTest {
-        settings.putLong(BreedModel.DB_TIMESTAMP_KEY,currentTimeMillis())
+    fun staleDataCheckTest() = runTest {
+        settings.putLong(BreedModel.DB_TIMESTAMP_KEY, currentTimeMillis())
         assertFalse(ktorApi.jsonRequested)
         model.getBreedsFromNetwork()
         assertFalse(ktorApi.jsonRequested)
     }
 
     @Test
-    fun `Update Breed Favorite`() = runTest {
+    fun updateFavoriteTest() = runTest {
         model.getBreedsFromNetwork()
+        itemDataSummary.await(500)
```

Comment:
Should we be asserting on the value in here? If not it can probably be typed `Unit`

Reply:
It definitely could be unit as it stands, but feel like it doesn't hurt to leave it as is and potentially be more useful for future tests